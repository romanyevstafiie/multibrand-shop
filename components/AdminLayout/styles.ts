import styled from 'styled-components';

export const LayoutContainer = styled.section`
  display: grid;
  grid-template-columns: 1fr 8fr;
  min-height: 100vh;
`;

export const ChildrenWrapper = styled.main`
  background-color: ${({ theme }) => theme.dashboardGrayBg};
`;
