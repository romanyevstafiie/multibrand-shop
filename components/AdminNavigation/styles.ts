import styled from 'styled-components';

export const StyledAdminNavigation = styled.aside`
  display: flex;
  align-items: center;
  justify-content: center;
  position: sticky;
  top: 0;
  padding: 5rem 1rem;

  li {
    margin: 1rem 0;
  }
`;
