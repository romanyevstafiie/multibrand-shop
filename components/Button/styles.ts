import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
