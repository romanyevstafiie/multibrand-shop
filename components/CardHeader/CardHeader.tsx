import React from 'react';
import styled from 'styled-components';

export interface Props {
  headerText: string;
}
/**
 * Reusable Card Header component
 *@function CardHeader
 *@param {string} headerText - Text content of Card Header
 *@returns {JSX.Element} - Rendered CardHeader component (
 */
const CardHeader = ({ headerText }: Props): JSX.Element => {
  return <StyledHeader>{headerText}</StyledHeader>;
};

const StyledHeader = styled.h3`
  font-size: 1.25rem;
  text-transform: capitalize;
  margin-bottom: 1rem;
`;

export default CardHeader;
