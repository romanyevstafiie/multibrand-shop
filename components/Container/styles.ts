import styled from 'styled-components';

export const StyledContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
  padding: 0 1rem;

  a {
    text-decoration: none;
  }
`;
