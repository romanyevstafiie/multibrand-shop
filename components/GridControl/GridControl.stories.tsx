import { Story } from '@storybook/react';
import React from 'react';
import GridControl, { Props } from './GridControl';

export default {
  title: 'Products/Grid Control Bar',
  component: GridControl,
};

const Template: Story<Props> = (args) => <GridControl {...args} />;

export const NavSearch = Template.bind({});
NavSearch.args = { header: 'All Products', productAmount: 777 };
