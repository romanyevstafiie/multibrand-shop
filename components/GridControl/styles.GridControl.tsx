import styled from 'styled-components';

export const Container = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  margin-bottom: 1rem;
  background: ${({ theme }) => theme.secondaryBg};
  border: 1px solid ${({ theme }) => theme.borderGrayColor};
  border-radius: ${({ theme }) => theme.borderRadius};

  h3 {
    margin: 0;
  }

  p {
    color: ${({ theme }) => theme.lightGray};
  }

  @media ${({ theme }) => theme.media.tablet} {
    padding: 1rem 2rem;
    h3 {
      display: none;
    }
  }
  @media ${({ theme }) => theme.media.phone} {
    .product__amount {
      display: none;
    }

    .items__per__page {
      display: none;
    }
  }
`;

export const SortWrapper = styled.div`
  position: relative;
  color: ${({ theme }) => theme.lightGray};
  cursor: pointer;

  &:hover {
    color: ${({ theme }) => theme.secondaryGolden};
  }

  @media ${({ theme }) => theme.media.phone} {
    span {
      display: none;
    }
  }
`;

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 1rem;

  .icon {
    cursor: pointer;
    font-size: 1.3rem;

    &:hover {
      color: ${({ theme }) => theme.secondaryGolden};
    }
  }

  .icon-list {
    font-size: 1.55rem;
  }
`;
