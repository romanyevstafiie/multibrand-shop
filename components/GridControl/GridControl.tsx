import React from 'react';
import { BsGrid3X3GapFill, BsListUl } from 'react-icons/bs';
import PageHeader from '../PageHeader/PageHeader';
import * as Styled from './styles.GridControl';

export interface Props {
  header: string;
  productAmount: number;
  changeLayout: (val: 'grid' | 'vertical') => void;
}
/**
 *@function GridControl
 *@param {string} header - Grid control header
 *@returns {JSX.Element} - Rendered GridControl component
 */
const GridControl = ({
  header,
  productAmount,
  changeLayout,
}: Props): JSX.Element => {
  // TODO: change paragraphs to controllable <details></details>
  // TODO: Implement sorting and pagination
  return (
    <Styled.Container>
      <PageHeader headerText={header} />
      <p>Showing {productAmount} products</p>
      <p>Display 24 per page</p>
      <p>Sort by: Alphabetically, A-Z</p>
      <Styled.IconContainer>
        <p>View</p>
        <BsGrid3X3GapFill
          className="icon"
          onClick={() => changeLayout('grid')}
        />
        <BsListUl className="icon" onClick={() => changeLayout('vertical')} />
      </Styled.IconContainer>
    </Styled.Container>
  );
};
export default GridControl;
