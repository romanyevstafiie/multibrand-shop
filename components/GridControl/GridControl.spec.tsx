import { shallow } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import GridControl from './GridControl';
import * as Styled from './styles.GridControl';
import CardHeader from '../CardHeader/CardHeader';
import renderer from 'react-test-renderer';
import { withReduxAndStyledProviders } from '../../test/testUtils';
import Filters from '../Filters/Filters';

describe('Navbar Header Search', () => {
  const wrapper = shallow(
    <GridControl
      header="tomato"
      changeLayout={() => console.log('tomato')}
      productAmount={25}
    />
  );

  it('should render with no errors', () => {
    expect(wrapper);
    expect(wrapper.length).toBe(1);
  });

  it('should render amount of products with no error', () => {
    const container = wrapper.find(Styled.Container);
    const productAmount = container.find('p');
    expect(productAmount.first().text()).toBe('Showing 25 products');
  });
  it('should render correct header text', () => {
    const header = wrapper.find(CardHeader);
    expect(header.length).toBe(1);
  });

  describe('SNAPSHOT', () => {
    test('it applies default styles', () => {
      const tree = renderer
        .create(withReduxAndStyledProviders(<Filters />))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
