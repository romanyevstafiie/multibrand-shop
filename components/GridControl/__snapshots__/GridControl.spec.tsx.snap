// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Header Search SNAPSHOT it applies default styles 1`] = `
.c1 {
  font-size: 1.25rem;
  text-transform: capitalize;
  margin-bottom: 1rem;
}

.c4 {
  padding: 0.25rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(4,1fr);
  gap: 0.5rem;
}

.c6 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: beige;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c7 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: black;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c8 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: blue;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c9 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: brown;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c10 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: gold;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c11 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: green;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c12 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: grey;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c13 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: ivory;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c14 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: multi;
  background: linear-gradient( to right,red,orange,yellow,green,blue,indigo,violet );
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c15 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: orange;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c16 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: pink;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c17 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: purple;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c18 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: red;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c19 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: silver;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c20 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: white;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c21 {
  cursor: pointer;
  background: transparent;
  border: none;
  height: 2.25rem;
  width: 2.25rem;
  background: yellow;
  margin: 0;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c2 {
  padding: 0.25rem;
}

.c3 {
  cursor: pointer;
  background: transparent;
  border: 1px solid rgba(143,143,143,0.1);
  border-radius: 0.5rem;
  color: #ffffff;
  padding: 0.5rem 1rem;
  margin: 0.25rem;
  -webkit-animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
  animation: dfiLNz 0.6s cubic-bezier(0.23,1,0.32,1) both;
}

.c3:hover {
  color: #ffc130;
  background-color: #1b1b1b;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 20%;
  height: 35%;
  padding: 1.25rem;
  overflow: hidden;
  background: #151515;
  border: 1px solid rgba(143,143,143,0.1);
  border-radius: 0.5rem;
}

@media (max-width:768px) {
  .c0 {
    width: 100%;
  }
}

<aside
  className="c0"
>
  <h3
    className="c1"
  >
    Filters
  </h3>
  
  
  
  
  <div
    className="c2"
  >
    <details
      className="stylesFilterTextOption__FilterOption-sc-9ak2es-1"
    >
      <summary>
        Brand
      </summary>
      <button
        className="c3"
        onClick={[Function]}
      >
        Adidas
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Air Jordan
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Asics
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Diadora
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Converse
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Native
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Yeezy
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Supreme
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        Timberland
      </button>
    </details>
  </div>
  <div
    className="c4"
  >
    <details
      className="stylesFilterColorOptions__FilterOption-sc-ka1lzr-1"
    >
      <summary>
        Color
      </summary>
      <div
        className="c5"
      >
        <button
          className="c6"
          onClick={[Function]}
        />
        <button
          className="c7"
          onClick={[Function]}
        />
        <button
          className="c8"
          onClick={[Function]}
        />
        <button
          className="c9"
          onClick={[Function]}
        />
        <button
          className="c10"
          onClick={[Function]}
        />
        <button
          className="c11"
          onClick={[Function]}
        />
        <button
          className="c12"
          onClick={[Function]}
        />
        <button
          className="c13"
          onClick={[Function]}
        />
        <button
          className="c14"
          onClick={[Function]}
        />
        <button
          className="c15"
          onClick={[Function]}
        />
        <button
          className="c16"
          onClick={[Function]}
        />
        <button
          className="c17"
          onClick={[Function]}
        />
        <button
          className="c18"
          onClick={[Function]}
        />
        <button
          className="c19"
          onClick={[Function]}
        />
        <button
          className="c20"
          onClick={[Function]}
        />
        <button
          className="c21"
          onClick={[Function]}
        />
      </div>
    </details>
  </div>
  <div
    className="c2"
  >
    <details
      className="stylesFilterTextOption__FilterOption-sc-9ak2es-1"
    >
      <summary>
        Department
      </summary>
      <button
        className="c3"
        onClick={[Function]}
      >
        kid
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        men
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        women
      </button>
    </details>
  </div>
  <div
    className="c2"
  >
    <details
      className="stylesFilterTextOption__FilterOption-sc-9ak2es-1"
    >
      <summary>
        Size
      </summary>
      <button
        className="c3"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        1.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        2.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        3.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        4.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        5.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        6.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        7.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        8.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        9.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        10
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        10.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        11
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        11.5
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        12
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        13
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        14
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        15
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        onesize
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        MISC
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        s
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        m
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        l
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        xl
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        xxl
      </button>
      <button
        className="c3"
        onClick={[Function]}
      >
        xs
      </button>
    </details>
  </div>
</aside>
`;
